gameplay-controller.js

```diff
@@ -81,54 +96,77 @@ export default class GameplayController {
         const nextMilestoneTime = progressMilestones[nextMilestone];
         if (this.gameProgressState.time > nextMilestoneTime) {
             this.gameProgressState.milestone = nextMilestone;
+            this.controllerState.weaponSpawnedInMilestone = false;
         }
     }
     #spawnEnemy() {
-        const x = position[0] - spawnRadiusHalf + spawnRadius * Math.random();
-        const y = position[1] - spawnRadiusHalf + spawnRadius * Math.random();
+        const dir = [Math.random() - 0.5, Math.random() - 0.5];
+        const mSq = dir[0]**2 + dir[1]**2;
+        let x;
+        let y;
+        if (mSq < 0.001) {
+            x = spawnRadius;
+            y = 0;
+        } else {
+            const m = mSq**0.5;
+            x = dir[0] * spawnRadius / m;
```

Comment:
**praise**: Excellent implementation of highscore tracking, which enhances replayability.